Assets/MixedRealityToolkit/MSBuild/Scripts/PluginAssemblyInfo.cs

```diff
@@ -0,0 +1,287 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#if UNITY_EDITOR
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.IO;
+using System.Linq;
+using UnityEditor;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.MSBuild
+{
+    /// <summary>
+    /// Type of Plugin.
+    /// </summary>
+    public enum PluginType
+    {
+        /// <summary>
+        /// A .NET DLL.
+        /// </summary>
+        Managed,
+
+        /// <summary>
+        /// A native (C++) dll.
+        /// </summary>
+        Native
+    }
+
+    /// <summary>
+    /// This is the information for the plugins in the Unity project.
+    /// </summary>
+    public class PluginAssemblyInfo : ReferenceItemInfo
+    {
+        /// <summary>
+        /// Ges the type of Plugin.
+        /// </summary>
+        public PluginType Type { get; }
+
+        /// <summary>
+        /// Gets the path relative to the assets folder.
+        /// </summary>
+        public string AssetsRelativePath { get; }
+
+        /// <summary>
+        /// Gets whether this plugin is auto referenced.
```

Comment:
>auto referenced. [](start = 40, length = 16)

What does this mean?

Reply:
Ah, is this unity "auto reference" concept?

---
In reply to: [302781511](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/5215#discussion_r302781511) [](ancestors = 302781511)

Reply:
Yep, will add comment.

---
In reply to: [302781656](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/5215#discussion_r302781656) [](ancestors = 302781656,302781511)